### ISSUE TITLE
Profile picture and user name still visible dispite user has been suspended/deleted

### ISSUE THREAD

**Issue by Nakaner** [openstreetmap/openstreetmap-website#3158]
I received a spam personal message via osm.org today and wanted to report the sender. The sender account was deleted or suspended already but I can still see the profile picture and user name when I view the message on osm.org.

Message ID: 945465

User name: Camilla Switzer

**Comment by tomhughes** [comment]
Yes messages sent by a deleted user are still visible to the recipient at the moment.

I'm in two minds about it - hiding them might be confusing if messages you had received suddenly vanished because the sender chose to delete their account and a user-user message can be viewed as a kind of joint property which is why we have two separate delete flags for sender and recipient.

**Comment by jidanni** [comment]
Well at Facebook and Google they vacuum up all traces of a spam attack.
The only place Facebook can't reach would be an old notification sitting in the user's Gmail Inbox.

**Comment by gravitystorm** [comment]
> I can still see the profile picture and user name when I view the message on osm.org

I want to double-check this. [When a user is deleted](https://github.com/openstreetmap/openstreetmap-website/blob/2d111c0f34613779bb730f1d0a823fa48e9d8439/app/models/user.rb#L244-L260) their avatar is removed and their display_name changed. So in that case, if you view the message on osm.org you should not see the profile picture any more. (You'll still see it on the notification email, since we attach the image to the email when sending).

Can you confirm whether or not this is the case? If you can still see them, then something has gone wrong with deleting the users, and that needs to be fixed separately.

If this topic is just about removing messages from deleted senders, then we should update the description accordingly.

**Comment by tomhughes** [comment]
A full delete would do that yes but we don't normally do that for spammers - we do what the site calls "hide" which just sets the user status to deleted without purging other data.

That's a deliberate choice on my part, to preserve information about the spammers so we can analyse things like what IP addresses and email domains they are using.
